athena_cli.py

```diff
@@ -284,14 +309,23 @@ def get_query_results(self, execution_id):
             results = None
             paginator = self.athena.get_paginator('get_query_results')
             page_iterator = paginator.paginate(
-                QueryExecutionId=execution_id
+                QueryExecutionId=execution_id,
+                PaginationConfig={'PageSize':1000}
             )
 
-            for page in page_iterator:
-                if results is None:
-                    results = page
-                else:
-                    results['ResultSet']['Rows'].extend(page['ResultSet']['Rows'])
+            pages = iter(page_iterator)
+            first_page = next(pages) # get first page so we can retrieve metadata for header
+
+            headers = list(h['Name'] for h in first_page['ResultSet']['ResultSetMetadata']['ColumnInfo'])
+            first_row = None if len(first_page['ResultSet']['Rows']) == 0 else list(self.get_col_value(col) for col in first_page['ResultSet']['Rows'][0]['Data'])
+            rows = self.yield_rows(first_page, pages)
+
+            # certain requests return the header as the first row, so skip it
+            if first_row == headers:
+                next(rows)
+
+            return (headers, rows)
```

Comment:
This `return` has made the code after this `try`/`exception` block unreachable.

Reply:
Whoops. This should be fixed now